### PR TITLE
Add dcw alias (E) to Figure.coast

### PR DESCRIPTION
**Pull request by willschlitzer**
Add `paint_country` as an alias for the `-E` parameter for `Figure.coast`. Add tests for the new parameter.

### CODE REVIEW

**Comment by seisman** [comment]
I'm not sure `paint_country` is a good name for this parameter. [GMT.jl uses `dcw`](https://www.generic-mapping-tools.org/GMT.jl/latest/coast/), but I don't like that one.

File: pygmt/base_plotting.py [hunk 1]
```diff
@@ -135,6 +136,16 @@ def coast(self, **kwargs):
...
+        paint_country : str or list
+            ``code1,code2,…[+l|L][+gfill][+ppen][+z]``
+            Select painting or dumping country polygons from the Digital Chart of the
+            World. Append one or more comma-separated countries using the 2-character
+            ISO 3166-1 alpha-2 convention. To select a state of a country
+            (if available), append .state, (e.g, US.TX for Texas).  To specify a whole
+            continent, prepend = to any of the continent codes (e.g. =EU for Europe).
+            Append **+p**pen to draw polygon outlines (default is no outline)
+            and **+g**fill to fill them (default is no fill). To apply different
+            settings to different countries, pass a list of string arguments.
```
Comment (seisman) [inline pygmt/base_plotting.py:148]: Some suggestions:

1. Add links for the "Digital Chart of the World" and "ISO 3166-1 alpha-2 convention"
2. **+z** is not explained.
3. **+l|L** is not explained.

**Comment by willschlitzer** [comment]
I'm not sure what a good alias would be; I felt like `paint_country` is awkward but I wanted the parameter to be descriptive (which I don't think something like `dcw` is). I'll give it some thought on alternatives.

**Comment by willschlitzer** [comment]
I think `dcw` may be the best option. While It's doesn't really explain what the argument is, I think it looks cleaner than `paint_country`.

File: pygmt/base_plotting.py (hunk 1 above)
Comment (willschlitzer) [reply pygmt/base_plotting.py]: Changes made as suggested; I'll admit I don't have a good grasp on what **+z** and **+l** do, so it's copied from the GMT documentation.

**Comment by willschlitzer** [comment]
I'm not sure why this failed almost all of its tests; I'm having no issues when I run `pytest --mpl-generate-path=baseline pygmt/tests/test_coast.py`

**Comment by seisman** [comment]
> I'm not sure why this failed almost all of its tests; I'm having no issues when I run `pytest --mpl-generate-path=baseline pygmt/tests/test_coast.py`

That's because https://github.com/GenericMappingTools/pygmt/pull/769#issuecomment-753479145. Now it should work if you update your branch.

**Comment by willschlitzer** [comment]
@seisman Thanks for the help; I updated the branch

File: pygmt/base_plotting.py [hunk 2]
```diff
@@ -135,6 +136,23 @@ def coast(self, **kwargs):
         shorelines : str
             ``'[level/]pen'``
             Draw shorelines [Default is no shorelines]. Append pen attributes.
+        dcw : str or list
+            ``code1,code2,…[+l|L][+gfill][+ppen][+z]``
```
Comment (seisman) [inline pygmt/base_plotting.py:140]: Please update the syntax following #631.

File: pygmt/base_plotting.py [hunk 3]
```diff
@@ -135,6 +136,23 @@ def coast(self, **kwargs):
...
+            Select painting or dumping country polygons from the
+            `Digital Chart of the World
+            <https://en.wikipedia.org/wiki/Digital_Chart_of_the_World>`__.
+            Append one or more comma-separated countries using the 2-character
+            `ISO 3166-1 alpha-2 convention
+            <https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2>`__.
+            To select a state of a country (if available), append
+            .state, (e.g, US.TX for Texas).  To specify a whole continent, prepend = to
```
Comment (seisman) [inline pygmt/base_plotting.py:148]:
```suggestion
            .*state*, (e.g, US.TX for Texas).  To specify a whole continent, prepend = to
```

File: pygmt/base_plotting.py [hunk 4]
```diff
@@ -135,6 +136,23 @@ def coast(self, **kwargs):
...
+            any of the continent codes (e.g. =EU for Europe).
+            Append **+p**pen to draw polygon outlines (default is no outline)
```
Comment (seisman) [inline pygmt/base_plotting.py:150]:
```suggestion
            Append **+p**\ *pen* to draw polygon outlines (default is no outline)
```

File: pygmt/base_plotting.py [hunk 5]
```diff
@@ -135,6 +136,23 @@ def coast(self, **kwargs):
...
+            and **+g**fill to fill them (default is no fill). Append **+l**|**+L**
```
Comment (seisman) [inline pygmt/base_plotting.py:151]:
```suggestion
            and **+g**\ *fill* to fill them (default is no fill). Append **+l**\ |**+L**
```

File: pygmt/base_plotting.py [hunk 6]
```diff
@@ -135,6 +136,23 @@ def coast(self, **kwargs):
...
+            to *=continent* to only list countries in that continent; repeat if more
+            than one continent is requested. Append **+z** to place the country code in
+            the segment headers via -Zcode settings.To apply different settings to
```
Comment (seisman) [inline pygmt/base_plotting.py:154]:
```suggestion
            the segment headers via **-Z**\ *code* settings.To apply different settings to
```

File: pygmt/base_plotting.py [hunk 7]
```diff
@@ -141,6 +142,25 @@ def coast(self, **kwargs):
...
+            *code1,code2,…*\ [**+l**\ |**L**\ ][**+g**\ *fill*\ ]
+            [**+p**\ *pen*\ ][**+z**]
+            Select painting or dumping country polygons from the
+            `Digital Chart of the World
+            <https://en.wikipedia.org/wiki/Digital_Chart_of_the_World>`__.
+            Append one or more comma-separated countries using the 2-character
+            `ISO 3166-1 alpha-2 convention
+            <https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2>`__.
+            To select a state of a country (if available), append
+            .*state*, (e.g, US.TX for Texas).  To specify a whole continent,
```
Comment (seisman) [inline pygmt/base_plotting.py:155]:
```suggestion
            .\ *state*, (e.g, US.TX for Texas).  To specify a whole continent,
```

File: pygmt/base_plotting.py (hunk 2 above)
Comment (willschlitzer) [reply pygmt/base_plotting.py]: Think I fixed it.

File: pygmt/base_plotting.py [hunk 8]
```diff
@@ -141,6 +142,25 @@ def coast(self, **kwargs):
         shorelines : str
             ``'[level/]pen'``
             Draw shorelines [Default is no shorelines]. Append pen attributes.
+        dcw : str or list
+            *code1,code2,…*\ [**+l**\ |**L**\ ][**+g**\ *fill*\ ]
```
Comment (seisman) [inline pygmt/base_plotting.py:146]:
```suggestion
            *code1,code2,…*\ [**+l**\|\ **L**\ ][**+g**\ *fill*\ ]
```

File: pygmt/base_plotting.py [hunk 9]
```diff
@@ -141,6 +142,25 @@ def coast(self, **kwargs):
...
+            <https://en.wikipedia.org/wiki/Digital_Chart_of_the_World>`__.
+            Append one or more comma-separated countries using the 2-character
+            `ISO 3166-1 alpha-2 convention
+            <https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2>`__.
+            To select a state of a country (if available), append
+            .\ *state*, (e.g, US.TX for Texas).  To specify a whole continent,
+            prepend **=** to any of the continent codes (e.g. =EU for Europe).
+            Append **+p**\ *pen* to draw polygon outlines
+            (default is no outline) and **+g**\ *fill* to fill them
+            (default is no fill). Append **+l**\ |**+L** to *=continent* to
```
Comment (seisman) [inline pygmt/base_plotting.py:159]:
```suggestion
            (default is no fill). Append **+l**\|\ **+L** to *=continent* to
```

**Review by seisman — APPROVED** [review]

**Pull request merged**